src/rooms/rooms.service.ts

```diff
@@ -119,14 +124,29 @@ else{
     async getHotelDetail(id:number):Promise<any>{
         const hotel = await this.facilityRepository.findOne({id});
         const room = await this.roomRepository.find({facility:hotel,status:RoomStatus.AVAILABLE})
+        var final=[]
```

Comment:
Use Relevant Variable names, "final" doesn't  give clarity to the use of the variable.